core/src/test/java/com/linecorp/armeria/common/stream/StreamMessageTest.java

```diff
@@ -306,6 +307,13 @@ void writeToFile(@TempDir Path tempDir) throws IOException {
         }
     }
 
+    @Test
+    void noopSubscribe() {
+        final StreamMessage<Integer> source = StreamMessage.of();
+        final CompletableFuture<Void> future = source.peek(i -> {}).subscribe();
```

Comment:
nit; What do you think of also adding the peeked values to an accumulator in the `peek` callback and verifying the result?

Reply:
Thank you for the good idea I'll apply it!